repose-aggregator/tests/test-bundles/filter-one/src/main/scala/org/openrepose/filters/core/filterone/ClassLoaderServletRequestWrapper.scala

```diff
@@ -19,12 +19,17 @@
  */
 package org.openrepose.filters.core.filterone
 
+import javax.servlet.http.{HttpServletRequest, HttpServletRequestWrapper}
+
+import org.apache.commons.lang3.StringUtils
+import org.openrepose.others.SimplicityDivine
+
 class ClassLoaderServletRequestWrapper(request: HttpServletRequest) extends HttpServletRequestWrapper(request) {
   override def getHeader(headerString: String): String = {
     println("\n\n\n")
     println(s"Requesting header ${headerString}")
     println("\n\n\n")
-    if (StringUtils.startsWith(headerString, "FOO")) {
+    if (StringUtils.startsWith(headerString, "FOO") && (super.getHeader(headerString) != null)) {
       return new SimplicityDivine().createBar
```

Comment:
Why all the return statements?
I get this was just a make it go task, so we might just want to make another task to actually make this production-like code.